code/datums/trading/goods.dm

```diff
@@ -287,13 +285,6 @@ Sells devices, odds and ends, and medical stuff
 	origin = "Robot Store"
 	possible_origins = list("AI for the Straight Guy", "Mechanical Buddies", "Bot Chop Shop", "Omni Consumer Projects")
 	possible_trading_items = list(
-		/obj/item/weapon/secbot_assembly/ed209_assembly = TRADER_THIS_TYPE,
```

Comment:
The removal seems a bit excessive, since you increased the number of items available to 5-8.
Might just remove the really useless ones and keep/add somewhat functional robots or maybe even certain damaged mechs

Reply:
The items here that were removed were only the bot assemblies. I don't think the robot store should be selling unfinished robots, so I removed it selling the assemblies. It still sells all the small robots like cleanbots, floorbots and medibots and etc. but there is no longer bloat from assemblies that no one ever is going to buy.

I don't think the robot store should sell mechs, it doesn't seem to fit in my head. Even more so damaged ones, they are a store, not a second hand discount wares store.

Do you still wish this changed?

Reply:
Furthermore, as much as I was staring at the merchant code, the amount of items at roundstart is effectively random because while yes the random amount of items spawned is 5-8, there also is the pool removal code which is activated by the same variable [force] as add-to-pool and is called upon everyone tick [every minute]. Since force is set to 1 from round start, the game will often remove one or more items from the merchant's inventory, but not always. So even with 5-8 'guaranteed items', I've had merchants spawn with only 3 items in inventory during testing.

Reply:
Not always because remove-from-pool code also has a rand in it.

Reply:
Well, I still think that adding something similar to cargo, where damaged mechs are offered could be interesting.
i.e. you can buy a damaged mech but only have to pay 50% of the original price.

Reply:
The main issue with that is that I don't know how to add a damaged mech spawn :/